### DISCUSSION TITLE
[@auth/core@0.24.0] account.providerAccountId is a random UUID now

### DISCUSSION THREAD

**Discussion by sillvva** [nextauthjs/next-auth#9879] [Help] [answered]
After updating `@auth/core` from version `0.21.0` to `0.24.0`, I noticed something strange.

In the `signIn` callback, the `account.providerAccountId` used to return `profile.sub` for Google OAuth or `profile.id` for Discord OAuth. These are the unique ids returned by those providers.

Now, `account.providerAccountId` is returning as a random UUID. Also, because the value is random, it is different with every login. This makes it impossible to query the existing account from the database with that value. For now, I'm using the value from `profile` to run the query instead, but it would be nice to know why it's returning a random value now for the account id.

Example: 
```
@auth/core@0.21.0: account.providerAccountId = 1081933111************
                                               ^ This is fixed
@auth/core@0.24.0: account.providerAccountId = 6450ea02-0285-4269-bf9d-880a4256d24d
                                               ^ This is random
```

My current workaround:
```ts
// google provider object
accountId: function (profile: Profile) {
	return profile.sub as string;
},

// signIn callback
async signIn({ account, profile }) {
	...
	const provider = providers.find((p) => p.id === account.provider);
	if (!provider) authErrRedirect(401, `Provider '${account.provider}' not found`);

	account.providerAccountId = provider.accountId(profile);
	const existingAccount = await prisma.account.findFirst({
		where: {
			provider: account.provider,
			providerAccountId: account.providerAccountId
		}
	});
	...
```

Is this intended? Is it a bug?

**Comment by yordis** [comment]
It seems to be a bug: https://github.com/nextauthjs/next-auth/pull/9932

**Reply by sillvva** [reply]
That does not appear to have fixed the issue. I am now on the latest `@auth/core@0.26.3` build and am still getting back a random uuid every single time I log in despite having an existing account in my db. Here is the `account` logged out from the signIn callback.

![CleanShot 2024-02-12 at 10 19 24@2x](https://github.com/nextauthjs/next-auth/assets/5138384/57abcccc-706f-484d-96a1-71d1eceac8a5)

The main issue is that it's random on every login, making it impossible to match against the db.

I'm not sure why it can't just continue to pull the same value from the profile that it used to pull, considering it's there and my above hack using that value instead still works.

**Reply by sillvva** [reply, marked as answer]
Looks like they're still working on it here: (issue) #9992 and (PR) #10008 

The fixed version: 0.27.0

**Reply by sillvva** [reply]
The PR in the selected answer has resolved the issue.
